File: src/main/java/me/andre111/d20server/scripting/expression/Condition.java
package me.andre111.d20server.scripting.expression;

public class Condition {
	private final Type type;
	private final int value;
	
	public Condition(Type type, int value) {
		this.type = type;
		this.value = value;
	}
	
	public boolean matches(int input) {
		switch(type) {
		case EQUAL:
			return input == value;
		case LESS_THAN:
			return input < value;
		case LESS_THAN_OR_EQUAL:
			return input <= value;
		case GREATER_THAN:
			return input > value;
		case GREATER_THAN_OR_EQUAL:
			return input >= value;
		}
		return false;
	}
	
	public static enum Type {
		EQUAL,
		LESS_THAN,
		LESS_THAN_OR_EQUAL,
		GREATER_THAN,
		GREATER_THAN_OR_EQUAL
	}
}

File: core/common/scripting/expression/condition.js
export class Condition {
    type;
    expression;

    currentValue;

    constructor(type, expression) {
        this.type = type;
        this.expression = expression;
    }

    eval(context) {
        this.currentValue = this.expression.eval(context).getValue();
    }

    matches(input) {
        switch(this.type) {
        case Condition.Type.EQUAL:
            return input == this.currentValue;
        case Condition.Type.LESS_THAN:
            return input < this.currentValue;
        case Condition.Type.LESS_THAN_OR_EQUAL:
            return input <= this.currentValue;
        case Condition.Type.GREATER_THAN:
            return input > this.currentValue;
        case Condition.Type.GREATER_THAN_OR_EQUAL:
            return input >= this.currentValue;
        }
        return false;
    }

    static Type = {
		EQUAL: 'EQUAL',
		LESS_THAN: 'LESS_THAN',
		LESS_THAN_OR_EQUAL: 'LESS_THAN_OR_EQUAL',
		GREATER_THAN: 'GREATER_THAN',
		GREATER_THAN_OR_EQUAL: 'GREATER_THAN_OR_EQUAL'
    }
}
